### ISSUE TITLE
EnsoGL shapes can receive mouse events after a parent has been unlinked from the display object hierarchy.

### ISSUE THREAD

**Issue by MichaelMauderer** [enso-org/enso#6600]
While investigating #6575, I found that sometimes shapes can receive mouse events, even though they should not be visible any more. 

There were a number of display objects parented like this: Display Object `A` -> Display Object `B` -> Display Object `C`.
`C` had an FRP that would show/hide further children based on mouse over/out events. 

When `A` was removed from the display, hierarchy, `C` would afterwards receive an `out` `over` `out` sequence of events. This would lead the children visible in the scene as artefacts. 

Unfortunately, there is no easy way to reproduce this issue (yet). The original bug randomly occurred when the visualization action bar was visible through hovering, and the visualization was disabled.

**Comment by wdanilo** [comment]
1. `This would lead the children visible in the scene as artefacts.` - I don't understand how `out`/`over` events can lead to children visible as artifacts.
2. Are we talking here about the same frame, or different frames? Do we update display object hierarchy anywhere in this story?

**Comment by Frizi** [comment]
It's not clear to me from the description if how the visual bug is connected with the mouse events.

If there are some supposedly destroyed shapes still visible on the screen, then those are also likely to be rendered in the object ID buffer, making them able to receive hover events. I don't understand though how the events could cause the elements to remain visible. More description is needed.

**Comment by MichaelMauderer** [comment]
> show/hide further children based on mouse over/out events.

This is not a general automatism, but custom logic here, that reacts to mouse events. Thus, these mouse events lead to `add_child`/`unset_parent` methods being called. 

For example, 
```
 eval model.shapes.mouse_over ( model.show() );
 eval model.shapes.mouse_out ( model.hide() );
```
(which is a simplified version of what is happening here :https://github.com/enso-org/enso/blob/1c894305974a71405977a5a3258e213885cc76e2/app/gui/view/graph-editor/src/component/visualization/container/action_bar.rs#L385 
https://github.com/enso-org/enso/blob/1c894305974a71405977a5a3258e213885cc76e2/app/gui/view/graph-editor/src/component/visualization/container/action_bar.rs#L390

---

> Are we talking here about the same frame, or different frames? Do we update display object hierarchy anywhere in this story?

@wdanilo  I'm not sure yet, this will need to be investigated.

**Comment by wdanilo** [comment]
@MichaelMauderer I have hard time understanding this issue. Do you have a debug scene showing it maybe?

**Comment by MichaelMauderer** [comment]
I believe this will be fixed with #7177, so I'm closing this.
